src/Databases/DDLDependencyVisitor.cpp

```diff
@@ -25,6 +28,7 @@ namespace
     /// Used to visits ASTCreateQuery and extracts the names of all tables explicitly referenced in the create query.
     class DDLDependencyVisitorData
     {
+        friend void tryVisitNestedSelect(const String & query, DDLDependencyVisitorData & data);
```

Comment:
Why not just make it a member function as other `visit*`?

Reply:
Because it uses `DDLDependencyVisitor`, and `DDLDependencyVisitor` is defined after `DDLDependencyVisitorData`